src/realm/obj.cpp

```diff
@@ -975,19 +975,22 @@ void Obj::to_json(std::ostream& out, size_t link_depth, const std::map<std::stri
         out << "\"" << name << "\":";
         prefixComma = true;
 
-        if (ck.get_attrs().test(col_attr_List)) {
-            if (type == col_type_LinkList) {
+        if (ck.is_list() || ck.is_set()) {
```

Comment:
Why not `is_collection()`? Does this not apply to dictionaries?

Reply:
No, Dictonary is handled next.